TranscriptionKit/Classes/Transcriber.swift

```diff
@@ -232,26 +181,251 @@ open class Transcriber: NSObject, AVAudioPlayerDelegate {
+
+            if error != nil || isFinal {
+                // Stop recognizing speech if there is a problem.
+                self.audioEngine.stop()
+                self.inputNode?.removeTap(onBus: 0)
+
+                self.recognitionRequest = nil
+                self.recognitionTask = nil
+
+                DispatchQueue.main.async {
+                    print("Ending transcription")
+                    self.transcribeStreamingClient.endTranscription()
+                }
+
+                self.delegate?.transcriberDidFinishRecognition?(self)
+            }
+        }
+        transcribeStreamingClient.startTranscriptionWSS(request)
+    }
+
+    private func record() throws {
+
+        guard let inputNode = inputNode else { return }
+
+        let recordingFormat = inputNode.outputFormat(forBus: 0)
+
+        var outDesc = AudioStreamBasicDescription()
+        outDesc.mSampleRate = recordingFormat.sampleRate
+        outDesc.mChannelsPerFrame = 1
+        outDesc.mFormatID = kAudioFormatFLAC
+
+        let framesPerPacket: UInt32 = 1152
+        outDesc.mFramesPerPacket = framesPerPacket
+        outDesc.mBitsPerChannel = 24
+        outDesc.mBytesPerPacket = 0
+
+        let convertFormat = AVAudioFormat(streamDescription: &outDesc)!
+        let converter = AVAudioConverter(from: recordingFormat, to: convertFormat)
+        self.converter = converter
+
+        let packetSize: UInt32 = 8
+        let bufferSize = 1 * packetSize
+
+        let audioFile = try AVAudioFile(forWriting: fileURL, settings: [AVFormatIDKey: kAudioFormatMPEG4AAC],
+                                        commonFormat: recordingFormat.commonFormat, interleaved: false)
+        inputNode.installTap(onBus: 0, bufferSize: bufferSize,
+                             format: recordingFormat) { [weak self] (buffer: AVAudioPCMBuffer, _: AVAudioTime) in
+
+            guard let self = self else { return }
+
+            let count = Int(buffer.frameLength)
+            guard count > 0 else { return }
+
+            let headers = [
+                ":content-type": "audio/flac",
+                ":message-type": "event",
+                ":event-type": "AudioEvent"
+            ]
+            
+            /*
+             * AVAudioPCMBuffer -> transcribeStreamingClient.send(data
+             */
+        
+            self.compressedBuffer = AVAudioCompressedBuffer(
+                format: convertFormat,
+                packetCapacity: packetSize,
+                maximumPacketSize: self.converter.maximumOutputPacketSize
+            )
+
+            let inputBlock: AVAudioConverterInputBlock = { (_, outStatus) -> AVAudioBuffer? in
+                outStatus.pointee = AVAudioConverterInputStatus.haveData
+                return buffer
+            }
+
+            var outError: NSError?
+            self.converter.convert(to: self.compressedBuffer!, error: &outError, withInputFrom: inputBlock)
+
+            let audioBuffer = buffer.audioBufferList.pointee.mBuffers
```

Comment:
... but here you're processing the original PCM buffer, not the new compressedBuffer...? I'm not familiar with AV conversions so maybe I'm just not understanding it properly...